packages/react-dom/src/events/DOMEventResponderSystem.js

```diff
@@ -534,9 +533,15 @@ function traverseAndHandleEventResponderInstances(
             const onEvent = responder.onEvent;
             if (onEvent !== null) {
               currentInstance = responderInstance;
-              responderEvent.responderTarget = ((target: any):
```

Comment:
The event was being mutated each time. If we want to reuse/freeze a single event object across all the responders perhaps we could move `responderTarget` to a context method like `getResponderTarget`.